src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -2626,17 +2636,23 @@ public static void setPaxosAutoRepairThresholdMB(int threshold)
 
     public static void setNativeTransportMaxConcurrentRequestsInBytesPerIp(long maxConcurrentRequestsInBytes)
     {
-        conf.native_transport_max_concurrent_requests_in_bytes_per_ip = maxConcurrentRequestsInBytes;
+        if (maxConcurrentRequestsInBytes < 0)
+            maxConcurrentRequestsInBytes = 0;
+
+        conf.native_transport_max_concurrent_requests_per_ip = DataStorageSpec.inBytes(maxConcurrentRequestsInBytes);
     }
 
     public static long getNativeTransportMaxConcurrentRequestsInBytes()
     {
-        return conf.native_transport_max_concurrent_requests_in_bytes;
+        return conf.native_transport_max_concurrent_requests.toBytes();
     }
 
     public static void setNativeTransportMaxConcurrentRequestsInBytes(long maxConcurrentRequestsInBytes)
     {
-        conf.native_transport_max_concurrent_requests_in_bytes = maxConcurrentRequestsInBytes;
+        if (maxConcurrentRequestsInBytes < 0)
+            maxConcurrentRequestsInBytes = 0;
```

Comment:
In the setter it was just assigning negative number and not really the default value... that smells like a bug but I will definitely verify it on the mailing list later with all other edge cases, as we spoke. In case of some tribal knowledge...

Reply:
And thanks for bringing it up!

Reply:
I'm almost certain that was a bug, but this has just not been a widely used feature, and even if it was, it's kind of a corner case. Either way, we can fix it here.